File: Data Structures & Algorithms/duplicate-integer/submission-4.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        HashSet<int> dupes = new HashSet<int>();

        for (int i = 0; i < nums.Length; i++) {
            dupes.Add(nums[i]);
        }

        return dupes.Count != nums.Length;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) < 2:
            return len(s)

        count = 0

        l = 0
        r = l + 1
        pos = {s[l]: l}
        while r < len(s):
            if pos.get(s[r]):
                pos = {}
                l += 1
                r += 1
            else:
                pos[s[r]] = r
                r += 1

            count = max(len(pos), count)

        return count